### DISCUSSION TITLE
How does HammerDb treat database errors?

### DISCUSSION THREAD

**Discussion by siddhsql** [TPC-Council/HammerDB#546] [Q&A] [answered]
https://www.hammerdb.com/docs/ch04s06.html
RAISEERROR impacts the behaviour of an individual virtual user on detecting a database error. If set to TRUE on detecting an error the user will report the error into the HammerDB console and then terminate execution. If set to FALSE the virtual user will ignore the error and proceed with executing the next transaction. 

question: are failed transactions still included in the NOPM and TPM metrics? i.e., there must be a counter(s) tracking the NOPM and TPM count. does that counter(s) get incremented even if there is an error returned by the database?

**Comment by sm-shaw** [comment, marked as answer]
All errors are caught. RAISEERROR determines whether the VU stops processing further transactions or continues.  Its all open source so you can see the counters by looking in the scripts. NOPM is a successful commit of the new order transaction. TPM counts commits and rollbacks that is database dependent.
